CMakeLists.txt

```diff
@@ -110,6 +112,38 @@ option(REQUIRE_PTHREADS "For non-Windows systems, set whether Csound will use th
 option(BUILD_DOCS "Build documentation (requires doxygen)" OFF)
 option(DEBUG_ERROR_ON_WARNING "For a debug build, error on a compiler# warning" OFF)
 option(BARE_METAL "To configure for bare metal build" OFF)
+option(ADD_COMMANDLINE_TESTS "Add command-line tests" ON)
+option(TRACK_COVERAGE "Track coverage" OFF)
+
+if (BUILD_TESTS)
+    # c tests require the static library
+    set(BUILD_STATIC_LIBRARY ON)
+endif()
+
+if (TRACK_COVERAGE)
+    check_cxx_compiler_flag(-g has_debug_info)
+    if (has_debug_optimization)
+        # disable optimizations for better coverage stats
+        add_compile_options(-Og)
+    endif()
+
+    # link option is required for the compiler option, so set and unset
+    set(CMAKE_REQUIRED_LINK_OPTIONS --coverage)
```

Comment:
Is this a G++/Clang flag? It won't work on Windows in that case

Reply:
Yes. I only added it temporarily so the line below would work on gcc/clang though. I remove it two lines later?